### PR TITLE
Figure 5 with larger text

### LINKED ISSUE #1740: Figure 5 Revisions for Publication
- Panels A and B need bigger legends and reorganization...
  - Reorient A legend to be on top of UMAP
  - Move panel B GSVA legend to vertical on the side
  - Explore rm'ing panel B legend and instead directly label the annotations at an angle (~60?)
- HR/P text for panel D
- Panel E can be bigger overall

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #1740 
**_Found another zenodo `\n` problem in this PR, fixed._** 😭  - Part of #1744

I'm opening this PR for Figure 5 before it's done to get some feedback, hence Draft. I've updated panels A-C so far, and it feels _really_ busy to me but I'm not sure that's entirely avoidable if we want larger legends! As I wrote in #1740, another approach for GSVA labeling could be to use the cancer group abbreviations to directly label the bar. Is this worth exploring?
Feedback very welcome! @jashapiro @jharenza

### CODE REVIEW

**Review by jashapiro — COMMENTED** [review]
I think having both of the cancer group labels at the bottom is making things quite confusing. I might move the one of the two sets of labels to the top?

Separately, I think there is something funky going on with your png conversions. The heatmap bars should be solid colors but have odd stripes in the PNG version. The PDF looks fine (Better with Acrobat than the mac PDF viewer, but that is common).

**Comment by sjspielman** [comment]
@jashapiro At a minimum I'm pretty sure I've fixed the PNG stripiness. Here's another partial compilation, again only first 3 panels. This time, the UMAP legend is below and the heatmap legend is above their respective plots. I do think it looks better, though still (unavoidably?) busy.

**Comment by jashapiro** [comment]
> @jashapiro At a minimum I'm pretty sure I've fixed the PNG stripiness. Here's another partial compilation, again only first 3 panels. This time, the UMAP legend is below and the heatmap legend is above their respective plots. I do think it looks better, though still (unavoidably?) busy.

I definitely like that better. One more little idea: We could eliminate the UMAP ticks and values... Just keep the UMAP1 & UMAP2 axis titles, but claim back a bit of space there and remove some clutter.

**Comment by jharenza** [comment]
Panel letters are missing here and looks like a cluster branch is also missing in B. Another possibility to increase pathway text in B could be to lengthen the heatmap and make C into 3 columns and 4 rows. This could also give you some more space to space out the legend in B.

**Comment by sjspielman** [comment]
This PR is now officially ready for real review. In https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1751/commits/9f09ad52e9d34d0d241b82e39a4820ec53e78bd4, I have a fully compiled updated Figure 5 for feedback! There is lots more space in between A/B which I think is helpful to make the overall figure less busy. While making this, I also explored swapping A and B (as in, heatmap on left and UMAP on right) to better line up with D and E, but that alignment did not work as well as I'd hoped, so I kept the panel arrangements as is.

File: figures/scripts/fig5-panels-gsva-umap.R [hunk 1]
```diff
@@ -260,28 +261,49 @@ gsva_heatmap <- Heatmap(
...
+
+histology_order <- names(annotation_colors_bh)
+
+display_group_legend_plot <- tibble::as_tibble(annotation_colors_bh, 
+                  rownames = "histology") %>%
+  mutate(histology = factor(histology, 
+                            levels = names(annotation_colors_bh))) %>%
+  ggplot() + 
+  aes(x = annotation_colors_bh, y = annotation_colors_bh, color = histology) + 
+  geom_point(size = 3.75) +
```
Comment (jashapiro) [inline figures/scripts/fig5-panels-gsva-umap.R:289]: These seem a touch large to me.

File: figures/scripts/fig5-panels-gsva-umap.R [hunk 2]
```diff
@@ -260,28 +261,49 @@ gsva_heatmap <- Heatmap(
...
+    legend.title = element_blank(),
+    legend.text = element_text(size = 8.5), 
+)
+  
+display_group_legend <- cowplot::get_legend(display_group_legend_plot)
+
+cowplot::save_plot(
+  umap_legend_pdf, 
+  cowplot::ggdraw(display_group_legend), 
+  base_width = 4.75, base_height = 1.25
```
Comment (jashapiro) [inline figures/scripts/fig5-panels-gsva-umap.R:304]: You are assembling this legend separately and joining illustrator, right? Can you squeeze them together a bit more? (Probably by making the UMAP a bit larger)

**Review by jashapiro — COMMENTED** [review]
I think this looks good, but maybe we can make the heatmap itself just a bit larger? And as noted, I think we can keep the UMAP a bit closer to its legend, and maybe just a bit smaller on those points (but I like that they are circles now!)

File: figures/scripts/fig5-panels-gsva-umap.R (hunk 2 above)
Comment (sjspielman) [reply figures/scripts/fig5-panels-gsva-umap.R:304]: Hm, maybe I should join with code... on it either way!

Comment (sjspielman) [reply figures/scripts/fig5-panels-gsva-umap.R]: Noting that given https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1753, I'm going to keep these as separate exports so the legend can still be popped into S7.

File: figures/scripts/fig5-panels-gsva-umap.R (hunk 1 above)
Comment (sjspielman) [reply figures/scripts/fig5-panels-gsva-umap.R]: Dropped in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1751/commits/15ec9447cc020cc0a368226e6b69a29abacfa3a5

**Review by jashapiro — APPROVED** [review]
LGTM!

**Comment by sjspielman** [comment]
All code changes here are in `figures/` so it doesn't run through CI. Can merge before checks.

**Pull request merged**